File: Leetcode/204. Count Primes/solution.py
class Solution:
    def countPrimes(self, n: int) -> int:
        counter = 0
        mem = [True] * n

        for num in range(2, n):
            if mem[num]:
                counter += 1

                for i in range(num * num, n, num):
                    mem[i] = False

        return counter

File: Leetcode/204. Count Primes/solution.rs
impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        let mut counter = 0;
        let mut mem = vec![true; n as usize];

        for num in 2..n {
            let index = num as usize;
            if mem[index] {
                counter += 1;
                for i in (num..n).step_by(index) {
                    mem[i as usize] = false
                }
            }
        }

        counter
    }
}
